optuna/_dataframe.py

```diff
@@ -0,0 +1,83 @@
+import collections
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Set
+from typing import Tuple
+
+import optuna
+from optuna._imports import try_import
+from optuna.trial._state import TrialState
+
+
+with try_import() as _imports:
+    # `trials_dataframe` is disabled if pandas is not available.
+    import pandas as pd  # NOQA
+
+# Required for type annotation in `Study.trials_dataframe`.
+if not _imports.is_successful():
+    pd = object  # type: ignore # NOQA
+
+
+def _trials_dataframe(
```

Comment:
The implementation of this function is just a copy from `Study.trials_dataframe`.